### PR TITLE
Refactor code to a package

### PR DESCRIPTION
**Pull request by dkuschmierz**
This PR has no code changes and refactors the script as a package. Executing the code will change from 

`python elf-size-analyze.py`

to 

`python -m elf_size_analyze`

This is the first step to publish the package to pypi. In the next PR, I will finish the step of moving to pypi and then also create an entry point to simply run the tool with 

`elf-size-analyze`

### CODE REVIEW

File: elf_size_analyze/color.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+def test__colors():
+    for attr in dir(Color):
+        if attr.isupper() and not attr.startswith('_'):
+            print(getattr(Color, attr) + 'attribute %s' % attr + Color.RESET)
+    sys.exit(0)
+
+
+#  test__colors()
```
Comment (jedrzejboczar) [inline elf_size_analyze/color.py:77]: As this is a separate file now we can provide a nice way of running a "color test" by executing this file as main (`python elf_size_analzye/color.py`), just add the following:
```python
def test_colors():
    for attr in dir(Color):
        if attr.isupper() and not attr.startswith('_'):
            print(getattr(Color, attr) + 'attribute %s' % attr + Color.RESET)
    sys.exit(0)

if __name__ == "__main__":
    test_colors()
```

File: elf_size_analyze/misc.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+"""
+Miscellaneous helper functions
+"""
+
+# construct python regex named group
+def g(name, regex):
```
Comment (jedrzejboczar) [inline elf_size_analyze/misc.py:6]: Now when it's in separate file the name could be more verbose (e.g. `named_group()`) and instead use shorthand when importing in other files, e.g.
```python
from elf_size_analyze.misc import named_group as g
```

File: elf_size_analyze/section.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                      hex(s.address), sizeof_fmt(s.size),
+                      ','.join(cls.Flag.to_string(f) for f in s.flags)]
+            lines.append(fields)
+        sizes = [max(len(l[i]) for l in lines) for i in range(6)]
+        h_fmt = '{:%d}   {:%d}   {:%d}   {:%d}   {:%d}   {:%d}' % (*sizes, )
+        fmt = '{:>%d}   {:%d}   {:%d}   {:>%d}   {:>%d}   {:%d}' % (*sizes, )
+        header = h_fmt.format('N', 'Name', 'Type', 'Addr', 'Size', 'Flags')
+        separator = '=' * len(header)
+        top_header = '{:=^{size}s}'.format(' SECTIONS ', size=len(separator))
+        print(Color.BOLD + top_header  + Color.RESET)
```
Comment (jedrzejboczar) [inline elf_size_analyze/section.py:149]: Missing import of `Color`, `python -m elf_size_analyze code.elf -P` results in error.

File: elf_size_analyze/tree_node.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
+"""
```
Comment (jedrzejboczar) [inline elf_size_analyze/tree_node.py:1]: I'd rename the file to `tree.py` as this is an implementation of a tree data structure. The class would stay as `TreeNode` as it represents a single node of a tree (and the root node is just the whole tree).

**Review by jedrzejboczar — CHANGES_REQUESTED** [review]
Looks good, just added a few minor comments.

File: elf_size_analyze/symbol_tree.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+        def __init__(self, depth=0, node=None, string=None, colors=None):
+            self.depth = depth
+            self.node = node
+            self.string = string
+            self.field_strings = []
+            self.colors = colors or []  # avoid creating one list shared by all objects
+
+        def print(self):
+            if len(self.colors) > 0:
+                print(sum(self.colors, Color()) + self.string + Color.RESET)
```
Comment (jedrzejboczar) [inline elf_size_analyze/symbol_tree.py:170]: Also missing import of `Color`.

File: elf_size_analyze/color.py (hunk 1 above)
Comment (dkuschmierz) [reply elf_size_analyze/color.py:77]: I would leave it for now, as I plan to add separate unit tests for the package later.

This is the plan what I want to do in the next steps (ordered by priority):

1. Refactor code with this PR
2. Create package and add GitHub action to publish on pypi after tagging
3. Add HTML output
4. Add pylint / flake8
5. Create UnitTests and add GitHub action on push

Comment (jedrzejboczar) [reply elf_size_analyze/color.py:77]: Ok, sounds cool

**Pull request merged**